### ISSUE TITLE
This Plugin doesn't work on Android 13

### ISSUE THREAD

**Issue by Kersei** [EinfachHans/cordova-plugin-advanced-imagepicker#53]
# Bug Report

**Current Behavior:**
I'm running the following code using ionic framework
`window.AdvancedImagePicker.present(
            {
                max: 10,
                maxCountMessage: this.translate.instant('Only 10 images can be uploaded at one time')
            },
            files => {
                files.forEach(fileObject => {
                        (<any>window).resolveLocalFileSystemURL(
                            fileObject.src,
                            fileEntry => {
                                fileEntry.file(
                                    file => {
                                        this.uploadFile(file);
                                    },
                                    error => {...}
                                );
                            },
                            err => {...}
                        );
                    }
                });
            },
           err => {...}
`
and when I test on an android phone with older version than 13 or ios it works just fine, but on Android 13 the plugin does nothing, it won't open and won't show any errors at all.

**Expected Behavior:**
the image picker should show on Android 13 or at least I should get an error telling me something is wrong.

**Steps to reproduce:**
try to use the plugin on an Android 13 device

**Environment information:**
- Cordova CLI version 11.0.0
- Cordova platform version
	- android 11.0.0
	- ios 6.2.0
- Plugins & versions installed in project (including this plugin)
`
com-badrit-base64 0.2.0 "Base64"
com-sarriaroman-photoviewer 1.3.0 "PhotoViewer"
com.darktalker.cordova.screenshot 0.1.6 "com.darktalker.cordova.screenshot"
cordova-android-support-gradle-release 3.0.1 "cordova-android-support-gradle-release"
cordova-plugin-androidx-adapter 1.1.3 "cordova-plugin-androidx-adapter"
cordova-plugin-badge 0.8.8 "Badge"
cordova-plugin-camera 6.0.1-dev "Camera"
cordova-plugin-customurlscheme 5.0.2 "Custom URL scheme"
cordova-plugin-device-motion 2.0.1 "Device Motion"
cordova-plugin-device 2.0.3 "Device"
cordova-plugin-file 7.0.0 "File"
cordova-plugin-filepath 1.6.0 "cordova-plugin-filepath"
cordova-plugin-fingerprint-aio 4.0.2 "FingerprintAllInOne"
cordova-plugin-geolocation 4.1.0 "Geolocation"
cordova-plugin-inappbrowser 5.0.0 "InAppBrowser"
cordova-plugin-ionic-keyboard 2.1.3 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 5.0.0 "cordova-plugin-ionic-webview"
cordova-plugin-local-notification 0.9.0-beta.4 "LocalNotification"
cordova-plugin-media-capture 4.0.0 "Capture"
cordova-plugin-nativegeocoder 3.5.1 "NativeGeocoder"
cordova-plugin-network-information 3.0.0 "Network Information"
cordova-plugin-advanced-imagepicker 1.6.2 "AdvancedImagePicker"
cordova-plugin-printer 0.8.0 "Printer"
cordova-plugin-safariviewcontroller 2.0.0 "SafariViewController"
cordova-plugin-screen-orientation 3.0.2 "Screen Orientation"
cordova-plugin-secure-storage-echo 5.1.1 "SecureStorage"
cordova-plugin-splashscreen 6.0.0 "Splashscreen"
cordova-plugin-statusbar 2.4.3 "StatusBar"
cordova-plugin-streaming-media 2.3.0 "StreamingMedia"
cordova-plugin-uniquedeviceid 1.3.2 "UniqueDeviceID"
cordova-plugin-zip 3.1.0 "cordova-plugin-zip"
cordova-sqlite-storage 6.0.0 "Cordova SQLite storage plugin - cordova-sqlite-storage plugin version"
es6-promise-plugin 4.2.2 "Promise"
onesignal-cordova-plugin 3.2.0 "OneSignal Push Notifications"
phonegap-plugin-barcodescanner-android12 8.1.2 "BarcodeScanner"
phonegap-plugin-mobile-accessibility 1.0.5-dev "Mobile Accessibility"
`
- Dev machine OS and version, e.g.
    - OSX
        ProductName:		macOS
        ProductVersion:		13.3
        BuildVersion:		22E252

**Comment by clarklight** [comment]
I am running on Android 13, and it works fine, but i am running on pure cordova, did you try to build a blank app and test out the image picker? I tested on Android 13, and 12 and they seems to be ok.

**Comment by RobinBrouwer** [comment]
Are you by any chance using targetSdkVersion 33? I had the same problem. The image selector wouldn't open and I didn't get any errors. I think it's some kind of permission error thing, because Android 13 has some changes regarding selecting media. You need to use READ_MEDIA_IMAGES and READ_MEDIA_VIDEO instead of READ_EXTERNAL_STORAGE. More info: https://developer.android.com/about/versions/13/behavior-changes-13.

I couldn't get it to work and a lot of other plugins stopped working (even after updating). I eventually set the targetSdkVersion to 32 and that fixed all my issues. Hope it helps.

**Comment by Kersei** [comment]
> 

yeah I'm using targetSdkVersion 33, unfortunately I cannot set it back to 32 due to some plugin restrictions, so I forked the plugin and made some changes https://github.com/Kersei/cordova-plugin-advanced-imagepicker

**Comment by EinfachHans** [comment]
@Kersei feel free to open a PR. I sadly dont have the time to support this plugin anymore

**Comment by clarklight** [comment]
@Kersei If it works well, can/should we merge the changes with Hans?

**Comment by Kersei** [comment]
> @Kersei If it works well, can/should we merge the changes with Hans?

It's working fine for my project but I don't think it's good enough for a PR

**Comment by JamieDickson** [comment]
updating the version of the tedimagepicker should be all that is needed.

#58

**Comment by kaeverens** [comment]
@JamieDickson is right - all you need to do is update the references from 1.2.8 to 1.3.2 and it works. thanks!

**Comment by degroundshaker** [comment]
@JamieDickson Having "ANDROID_IMAGE_PICKER_VERSION": "1.3.2" fixed it for me. On Android 13, native picker was being invoked. 1.3.2 fixed it for me.

**Comment by Mr-Anonymous** [comment]
I have the same issue. Can you tell me how you resolved this please? I use ionic and I tried changing the value from 1.2.8 to 1.3.2 in my config.xml so it looks like this but it doesn't change anything. The plugin is still not responding in Android 13 devices:

```
  <plugin name="cordova-plugin-advanced-imagepicker" spec="^1.6.2">
        <variable name="ANDROID_IMAGE_PICKER_VERSION" value="1.3.2" />
    </plugin>
```

Here is my ionic info:

```
Ionic:

   Ionic CLI         : 6.18.1 (/Users/me/.nvm/versions/node/v11.12.0/lib/node_modules/@ionic/cli)
   Ionic Framework   : ionic1 1.3.3
   @ionic/v1-toolkit : 1.0.9

Cordova:

   Cordova CLI       : 8.0.0
   Cordova Platforms : android 9.0.0, ios 6.1.1
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, (and 18 other plugins)

Utility:

   cordova-res                          : not installed globally
   native-run (update available: 2.0.0) : 1.7.1

System:

   Android SDK Tools : 26.1.1 (/Users/me/Library/Android/sdk)
   ios-sim           : 8.0.2
   NodeJS            : v11.12.0 (/Users/me/.nvm/versions/node/v11.12.0/bin/node)
   npm               : 6.14.7
   OS                : macOS Monterey
   Xcode             : Xcode 14.2 Build version 14C18

```

Can you help me please? Ever since I updated the min target sdk version to 33 due to Google policy requirement/nov 1 deadline, I started having this issue now.

**Comment by Mr-Anonymous** [comment]
> > 
> 
> yeah I'm using targetSdkVersion 33, unfortunately I cannot set it back to 32 due to some plugin restrictions, so I forked the plugin and made some changes https://github.com/Kersei/cordova-plugin-advanced-imagepicker

Hi @Kersei I tried to add the patch you have in your fork but it is not working for me. Can you tell me what cordova android version do you have please? I also tried adding `ANDROID_IMAGE_PICKER_VERSION` to version `1.4.2` which doesn't do anything. I am struggling to get this working in my Cordova Android@9.0.0 set-up. 

Can someone help me please if you have any solution for this SDK 33 related issue?

**Comment by Mr-Anonymous** [comment]
ok, I spent the whole day updating all of these and the recommended solution doesn't work:

1. Updated Cordova CLI to 12.0.0
2. Update Node to v18.18.2
3. Updated Java to 17
4. Updated Gradle
5. Installed Cordova Android 12.0.0
6. Removed all other conflicting and deprecated plugins.
7. Installed this `cordova-plugin-advanced-imagepicker`
8. Edited plugin.xml's `ANDROID_IMAGE_PICKER_VERSION` value to `1.4.2`.
9. Build the app

Still nothing happens. When clicked, the plugin is not triggered and the phone gallery doesn't open. I am now literally out of options. Any help from people who have this working will be a life saver and I will be hugely grateful! Thank you for reading this.

**Comment by clarklight** [comment]
@Mr-Anonymous Have you try building a brand new app with just this plugin? See if that works.

**Comment by Mr-Anonymous** [comment]
@clarklight Thank you for that. Ok, I tried that just now. I installed a new app on the exact same environment. Since I use an ionic 1 app, the dummy app was on the same version and same cordova/java/build tools set-up and I had the same issue in that. What I noticed is:

1. If I install the App on my older Android phone that has Android 10, it works fine.

2. If I install the App on my newer Android 13 phone, the plugin doesn't respond and there is no error either.

3. If I change the `android-targetSdkVersion` to less than 33, it works. But it doesn't work on API 33.

Hope this helps in some ways.

**Comment by clarklight** [comment]
@Mr-Anonymous mine worked but its on cordova, but i also had that same issue in the beginning. 

1.) Make sure you know there are 2 link to reference it from, 1 is this official one, the other one https://github.com/Kersei/cordova-plugin-advanced-imagepicker   you have to reference directly from that link

2.) I also tried adding in the permission plugin, and then requesting permission manually to identify if its a permission issue, because if it is, atleast still have an option to get it working. 

Hope it helps.

**Comment by Mr-Anonymous** [comment]
I was able to finally get this working. First of all, thank you so much for everyone who tried to help me.

What worked for me in the end was to change a few things in my environment set-up and after a lot of permutations and combinations, the following set-up works for my project:

1. I had to change my Java version from 17 to 11. (made a big difference for some reason)

2. Use Node v18.18.2 (don't think this matters much).

3. Use Cordova Android 12.0.0. (made a difference too. Even Android 10 worked but that was giving other issues with other plugins)

4. Use the cordova cli to build instead of using Android Studio. (this made a big difference too.)

Now with the above changes, I am finally able to get the plugin to work. 

Now I am using the official version of this plugin with that 1 added change to plugin.xml as mentioned here https://github.com/EinfachHans/cordova-plugin-advanced-imagepicker/pull/58

Thanks again! I am glad I am able to get this to work after 4 days!
